mis_builder/models/aep.py

```diff
@@ -0,0 +1,205 @@
+        if not mode:
+            mode = 'p'
+        elif mode == 's':
+            mode = 'e'
+        if account_codes.startswith('_'):
+            account_codes = account_codes[1:]
+        else:
+            account_codes = account_codes[1:-1]
+        account_codes = [a.strip() for a in account_codes.split(',')]
+        domain = domain or '[]'
+        domain = tuple(eval(domain))  # TODO: safe_eval
+        return field, mode, account_codes, domain
+
+    def parse_expr(self, expr):
+        """Parse an expression, extracting accounting variables.
+
+        Domains and accounts are extracted and stored in the map
+        so when all expressions have been parsed, we know what to query.
+        """
+        for mo in self.ACC_RE.finditer(expr):
+            field, mode, account_codes, domain = self._parse_mo(mo)
+            key = (domain, mode)
+            self._map[key].update(account_codes)
+
+    def done_parsing(self, account_domain):
+        # load account codes and replace account codes by account ids in _map
+        for key, account_codes in self._map.items():
+            self._load_account_codes(account_codes, account_domain)
+            account_ids = set()
+            for account_code in account_codes:
+                account_ids.update(self._account_ids_by_code[account_code])
+            self._map[key] = list(account_ids)
+
+    def get_aml_domain_for_expr(self, expr):
+        """ Get a domain on account.move.line for an expression.
+
+        Only accounting expression in mode 'p' and 'e' are considered.
+
+        Prerequisite: done_parsing() must have been invoked.
+
+        Returns a domain that can be used to search on account.move.line.
+        """
+        domains = []
+        for mo in self.ACC_RE.finditer(expr):
+            field, mode, account_codes, domain_partial = self._parse_mo(mo)
+            if mode == 'i':
+                continue
+            account_ids = set()
+            for account_code in account_codes:
+                account_ids.update(self._account_ids_by_code[account_code])
+            domain = [('account_id', 'in', tuple(account_ids))]
+            domain.extend(list(domain_partial))
+            if field == 'crd':
+                domain.append(('credit', '>', 0))
+            elif field == 'deb':
+                domain.append(('debit', '>', 0))
+            domain.extend(domain)
+            domains.append(expression.normalize_domain(domain))
+        return expression.OR(domains)
+
+    def do_queries(self, period_domain, period_domain_i, period_domain_e):
+        aml_model = self.env['account.move.line']
+        self._data = {}  # {(domain, mode): {account_id: (debit, credit)}}
+        for key in self._map:
+            self._data[key] = {}
+            domain, mode = key
+            if mode == 'p':
+                domain = list(domain) + period_domain
+            elif mode == 'i':
+                domain = list(domain) + period_domain_i
+            elif mode == 'e':
```

Comment:
I want to document i (= initial balance), e (= ending balance) and p (default if not set, = variation over the period). s did not translate well to english and I'd like to deprecate it.

Well, a migration is relatively hard and risky while the small backward compatibility hack here is simple.